robot/Cumulus/tests/browser/gift_entry/edit_template.robot

```diff
@@ -0,0 +1,139 @@
+&{custom}         Required=checked
+${amount}         50
+${msg}            Automation test
+
+*** Keywords ***
+Setup Test Data
+    &{contact} =     API Create Contact                 FirstName=${faker.first_name()}    LastName=${faker.last_name()}
+    Set suite variable    &{contact}
+    &{account} =     API Create Organization Account    Name=${faker.company()}
+    Set suite variable    &{account}
+    ${org_ns} =  Get Org Namespace Prefix
+    Set suite variable    ${org_ns}
+    ${ns} =  Get NPSP Namespace Prefix
+    Set suite variable    ${ns}
+
+
+*** Test Cases ***
+
+Edit GE Template And Verify Changes
+    [Documentation]                  This test case makes edits to Default gift entry template and verifies that edits are saved 
+    ...                              and available when a Single gift or batch gift are created. 
+ 
+    [tags]                           unstable     feature:GE          W-039559   
+    # Edit Default template to add some default values and add a new field to form                   
+    Go To Page                       Landing                       GE_Gift_Entry
+    Click Link                       Templates
+    Select Template Action           Default Gift Entry Template   Edit
+    Current Page Should Be           Template                      GE_Gift_Entry
+    Click Gift Entry Button          Next: Form Fields
+    Select Object Group Field        Account 1                     custom_acc_text 
+    Select Object Group Field        Opportunity                   Donation Imported
+    Fill Template Form                      
+    ...                              Account 1: custom_acc_text=&{custom}
+    ...                              Opportunity: Close Date=&{date}
+    ...                              Payment: Check/Reference Number=&{check}
+    ...                              Payment: Payment Method=&{method} 
+    Click Gift Entry Button          Save & Close
+    # Verify Default values are displayed on the Single Gift Form
+    Current Page Should Be           Landing                       GE_Gift_Entry
+    Click Gift Entry Button          New Single Gift
+    Current Page Should Be           Form                          Gift Entry
+    ${ui_date} =                     Get Current Date              result_format=%b %-d, %Y
+    ${date} =     Get Current Date    result_format=%Y-%m-%d
+    Verify Field Default Value
+    ...                              Donation Date=${ui_date}
+    ...                              Check/Reference Number=abc11233
+    ...                              Payment Method=Check
+    # Verify Single Gift form throws error when a required field is not filled, correct and save 
+    Fill Gift Entry Form
+    ...                              Donor Type=Account1
+    ...                              Existing Donor Organization Account=&{account}[Name]
+    ...                              Donation Amount=${amount}
+    Click Button                     Save
```

Comment:
Save step can be part of the Filling the gift entry form like the keyword can be  Populate_and_Save_Gift_Entry_Form

Reply:
there could be some other things that can be done on the form other than just filling the fields so I do not want to add save as part of it